src/web/components/StickyBottomBanner/BrazeBanner.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect, useState } from 'react';
+import * as emotion from 'emotion';
+import * as emotionCore from '@emotion/core';
+import * as emotionTheming from 'emotion-theming';
+import { onConsentChange } from '@guardian/consent-management-platform';
+import { getZIndex } from '@root/src/web/lib/getZIndex';
+import { Props as BrazeBannerProps } from '@guardian/braze-components';
+import { CanShowResult } from './bannerPicker';
+
+export const brazeVendorId = '5ed8c49c4b8ce4571c7ad801';
+
+type Props = {
+    meta: any;
+};
+
+const containerStyles = emotion.css`
+    position: fixed;
+    bottom: -1px;
+    width: 100%;
+    ${getZIndex('banner')}
+`;
+
+export const hasRequiredConsents = (): Promise<boolean> =>
+    new Promise((resolve) => {
+        onConsentChange(({ tcfv2, ccpa }) => {
+            const consentGivenUnderCcpa = ccpa && !ccpa.doNotSell;
+            const consentGivenUnderTcfv2 =
+                tcfv2 && tcfv2.vendorConsents[brazeVendorId];
+
+            resolve(!!(consentGivenUnderCcpa || consentGivenUnderTcfv2));
+        });
+    });
+
+export const canShow = async (
+    asyncBrazeUuid: Promise<null | string>,
+    isDigitalSubscriber: undefined | boolean,
+): Promise<CanShowResult> => {
```

Comment:
I don't know if this is useful or not, but for me a comment that encompasses the following steps in code is useful for understanding and sense-checking everything is captured. 

```suggestion
): Promise<CanShowResult> => {
// We can show a Braze banner if:
// - The Braze switch is on
// - We have a braze API key
// - The user is not a digital subscriber
// - We've received a Braze UUID from the API, given a user's ID Creds
// - The user has given Consent via CCPA or TCFV2
// - The Braze websdk appboy intialisation does not throw an error
// - The Braze app Boy subscription to in app message returns meta info
```

Reply:
Ah that's an interesting point. My approach usually is to name things in a way that makes the logic clear from the code (because comments get out of date). Perhaps adding this to the commit message would be useful 🤔

Reply:
Actually, having thought this over, because this list of conditions is quite long (and most are unlikely to change) I think that the comment is worth having :+1:

Reply:
> Actually, having thought this over, because this list of conditions is quite long (and most are unlikely to change) I think that the comment is worth having 👍

Yeah, I think that was my thinking, I had to go through quite a few steps to get to that point (it's not in one big if statement because of the early returns), so found it useful. But defo could go the other way if it gets out of date! Useful until it's not I guess 😆